massa-pos-worker/src/controller.rs

```diff
@@ -61,22 +62,23 @@ impl SelectorController for SelectorControllerImpl {
         lookback_rolls: BTreeMap<Address, u64>,
         lookback_seed: Hash,
     ) -> PosResult<()> {
-        {
-            // check status
-            let (_cache_cv, cache_lock) = &*self.cache;
-            cache_lock.read().as_ref().map_err(|err| err.clone())?;
-        }
-        {
-            // send command
-            let (cvar, lock) = &*self.input_data;
-            let mut data = lock.lock();
-            data.push_back(Command::DrawInput {
+        // check status
+        let (_cache_cv, cache_lock) = &*self.cache;
+        cache_lock.read().as_ref().map_err(|err| err.clone())?;
```

Comment:
It looks like the lock is still being held when sending on the channel, which could result in deadlock if:
1. the channel is full(blocking send)
2. the worker is blocked trying to acquire a write lock(see `process_draws_result`)

@damip

Reply:
good catch ! solved